summoner-tui/src/Summoner/Tui.hs

```diff
@@ -175,6 +182,17 @@ appNew dirs = App
                 then loopWhileInactive ev newForm
                 else continue newForm
 
+    -- Handles auto-fill of Prelude module name
+    autoFillPreludeModuleName
+        :: BrickEvent SummonForm e
+        -> KitForm e
+        -> SummonForm
+        -> EventM SummonForm (Next (KitForm e))
+    autoFillPreludeModuleName ev form preludeField = withForm ev (getUpdatedForm form preludeField) mkNewForm
+
+    getUpdatedForm :: KitForm e -> SummonForm -> KitForm e
+    getUpdatedForm form field = form
```

Comment:
You can take as an example this function:
https://github.com/kowainik/summoner/blob/de3b4cbfd85abd77ff86d0746ac81d3b6773651a/summoner-tui/src/Summoner/Tui.hs#L142-L143

You will just need to update another field instead.
Let me know if this helps.

Reply:
So, to update a field in the form you need to construct a whole new `SummonKit`, right?
In order to get the updated form, I changed `getUpdatedForm` to use a new `SummonKit` with all fields same except the `preludeModule`.
But I am getting an error
 ```
Variable not in scope:
      gitignore :: Lens.Micro.Type.Getting [Text] (KitForm e1) [Text]
 225 |                                                         , summonKitGitignore    = form ^. gitignore

```

Reply:
Oh, no. You don't need to. As you see in the example that I pasted in the previous comment you can use setters to update only fields that you need to update. In your case, you need to update `projectMetaPreludeModule` for which you should use `preludeModule` lens.

Reply:
I have pushed an implementation using the `preludeModule` lens to update the field with the value from `preludeName` field. But I am getting the following error on line 201:
```
src/Summoner/Tui.hs:201:16: error:
Couldn't match expected type ‘SummonKit’
                  with actual type ‘f0 SummonKit’
```

This seems like a noobish question, so I tried looking for a solution for quite a few days but could not find any relevant resources with a solution online.